src/io/flutter/run/OpenMemoryDashboardAction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.run;
+
+import com.intellij.ide.browsers.BrowserFamily;
+import com.intellij.ide.browsers.BrowserLauncher;
+import com.intellij.ide.browsers.WebBrowser;
+import com.intellij.ide.browsers.WebBrowserManager;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.util.Computable;
+import com.jetbrains.lang.dart.ide.runner.ObservatoryConnector;
+import icons.FlutterIcons;
+import io.flutter.FlutterBundle;
+import io.flutter.FlutterInitializer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+@SuppressWarnings("ComponentNotRegistered")
+public class OpenMemoryDashboardAction extends DumbAwareAction {
+
+  private final @NotNull ObservatoryConnector myConnector;
+  private final Computable<Boolean> myIsApplicable;
+
+  public OpenMemoryDashboardAction(@NotNull final ObservatoryConnector connector, @NotNull final Computable<Boolean> isApplicable) {
+    super(FlutterBundle.message("open.memorydashboard.action.text"), FlutterBundle.message("open.memorydashboard.action.description"),
+          FlutterIcons.OpenMemoryDashboard);
+    myConnector = connector;
+    myIsApplicable = isApplicable;
+  }
+
+  @Override
+  public void update(@NotNull final AnActionEvent e) {
+    e.getPresentation().setEnabled(myIsApplicable.compute());
+  }
+
+  @Override
+  public void actionPerformed(@NotNull final AnActionEvent e) {
+    FlutterInitializer.sendAnalyticsAction(this);
+
+    final String url = myConnector.getBrowserUrl();
+    if (url != null) {
+      openInAnyChromeFamilyBrowser(url + "#/memory-dashboard?editor=IntelliJ");
```

Comment:
My understanding is that `/ws` generally does a protocol upgrade to websockets, and other urls can just be handled by the web server; sounds like it would require changes to the vm's service server. It's something to consider - it's nice to have clean, short uls (/cc @lukechurch).

Re: the IntelliJ suffix, what do we get from the url param that we don't get from the info when the IDE registers itself w/ the service?